core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -208,23 +201,14 @@ public static Catalog loadCatalog(
    */
   public static Catalog buildIcebergCatalog(String name, Map<String, String> options, Configuration conf) {
     String catalogImpl = options.get(CatalogProperties.CATALOG_IMPL);
+    String catalogType = options.get(CatalogProperties.CATALOG_TYPE);
+    Preconditions.checkArgument(catalogImpl == null || catalogType == null,
+        "Cannot create catalog %s, both type and catalog-impl are set: type=%s, catalog-impl=%s",
+        name, catalogType, catalogImpl);
+
     if (catalogImpl == null) {
-      String catalogType = PropertyUtil.propertyAsString(options, ICEBERG_CATALOG_TYPE, ICEBERG_CATALOG_TYPE_HIVE);
-      switch (catalogType.toLowerCase(Locale.ENGLISH)) {
-        case ICEBERG_CATALOG_TYPE_HIVE:
-          catalogImpl = ICEBERG_CATALOG_HIVE;
-          break;
-        case ICEBERG_CATALOG_TYPE_HADOOP:
-          catalogImpl = ICEBERG_CATALOG_HADOOP;
-          break;
-        default:
-          throw new UnsupportedOperationException("Unknown catalog type: " + catalogType);
-      }
-    } else {
-      String catalogType = options.get(ICEBERG_CATALOG_TYPE);
-      Preconditions.checkArgument(catalogType == null,
-          "Cannot create catalog %s, both type and catalog-impl are set: type=%s, catalog-impl=%s",
-          name, catalogType, catalogImpl);
+      catalogType = (catalogType == null) ? CatalogType.HIVE.getTypeName() : catalogType;
```

Comment:
Nit: Can we add a log if neither of them are set and then mention that we will be using `hive`?

I know that this isn't a change from the present behavior of falling back to `hive`, but I have heard from a few users seeking support in the Iceberg slack who didn't fully setup their catalog properly and I think the log would benefit them.

Consider perhaps logging something like `Neither "type" nor "catalog-impl" were configured. Falling back to the default of "%s"`.

Also, perhaps consider making `CatalogType.Hive` marked as a default value somewhere? Usually with configs, we have the config name and then the default value as well. I think that would make the code more readable and in line with the other code in Iceberg.

Previously, the old code at least used the default value argument of `propertyAsString`.